plasmapy/formulary/parameters.py

```diff
@@ -161,7 +163,10 @@ def mass_density(
 @check_relativistic
 @validate_quantities(density={"can_be_negative": False})
 def Alfven_speed(
-    B: u.T, density: [u.m ** -3, u.kg / u.m ** 3], ion="p+", z_mean=None
+    B: u.T,
+    density: [u.m ** -3, u.kg / u.m ** 3],
+    ion: Optional[Particle] = None,
```

Comment:
I'm not sure if I did this annotation right (I'm pretty new to using type annotations). I followed the example of the `mass_density` function.

Also, question. Should the previous line actually be using `Union`? My IDE is complaining about using a list for the type annotation.
```suggestion
    density: Union[u.m ** -3, u.kg / u.m ** 3],
    ion: Optional[Particle] = None,
```

Reply:
This should still be `[u.m ** -3, ...]` since unit annotations aren't *type hint* annotations, strictly speaking, but are rather used by `plasmapy.utils.validate_quantities` and `astropy.units.quantity_input` to check that arguments have the correct units.  Last I heard, it's still an open issue in Astropy to allow unit annotations that also serve as type hint annotations (e.g., https://github.com/astropy/astropy/pull/10662).  For the moment, we should not use `Union` here, but I'm hoping that this inconsistency (that annotations are used for things besides type hints) eventually gets resolved.

Reply:
Once 10662 is merged then, for python 3.7+,`astropy.units.quantity_input ` can work with
```python
density: Union[Quantity[u.m ** -3], Quantity[u.kg / u.m ** 3]]
```
For python 3.6+ the manual construction would have to be used
```python
density: Union[Annotated[Quantity, u.m ** -3], Annotated[Quantity, u.kg / u.m ** 3]]
```